crates/red_knot_python_semantic/src/types.rs

```diff
@@ -377,15 +378,32 @@ impl<'db> FunctionType<'db> {
     /// annotated return type for this function, if any
     pub fn returns(&self, db: &'db dyn Db) -> Option<Type<'db>> {
         let definition = self.definition(db);
-        let DefinitionKind::Function(function_stmt_node) = definition.node(db) else {
-            panic!("Function type definition must have `DefinitionKind::Function`")
-        };
+        let function_stmt_node = Self::node(db, definition);
```

Comment:
I think this should just be `self.node(db)` -- we don't have to avoid calling `self.definition(db)` twice.

Reply:
Ah, good point; then we can simplify the call below to just be `self.definition(db)` inline as well.